firecrest/cli2/__init__.py

```diff
@@ -0,0 +1,997 @@
+#
+#  Copyright (c) 2019-2023, ETH Zurich. All rights reserved.
+#
+#  Please, refer to the LICENSE file in the root directory.
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+import json
+import logging
+import re
+import typer
+import yaml
+
+import firecrest as fc
+
+from firecrest import __app_name__, __version__
+from typing import List, Optional
+from enum import Enum
+
+from rich import box
+from rich.console import Console
+from rich.logging import RichHandler
+from rich.table import Table
+from rich.theme import Theme
+
+
+app = typer.Typer(
+    # Disable printing locals to avoid printing the value of local
+    # variables in order to hide secrets/password etc
+    pretty_exceptions_show_locals=False,
+)
+
+custom_theme = {
+    "repr.attrib_name": "none",
+    "repr.attrib_value": "none",
+    "repr.number": "none",
+}
+console = Console(theme=Theme(custom_theme))
+client: fc.v2.Firecrest = None  # type: ignore
+logger = logging.getLogger(__name__)
+
+
+def examine_exeption(e: Exception) -> None:
+    msg = f"{__app_name__}: Operation failed"
+    if isinstance(e, fc.ClientsCredentialsException):
+        msg += ": could not fetch token"
+    elif isinstance(e, fc.FirecrestException):
+        msg += ": a Firecrest client error has occurred"
+    # else:
+        # in case of FirecrestException and ClientsCredentialsException
+        # we don't need to log again the exception
+    msg += f": {e}"
+
+    console.print(f"[red]{msg}[/red]")
+
+
+def version_callback(value: bool):
+    if value:
+        console.print(f"FirecREST CLI Version: {__version__}")
+        raise typer.Exit()
+
+
+def config_parent_load_callback(ctx: typer.Context, param: typer.CallbackParam, value: str):
+    ctx.default_map = ctx.parent.default_map  # type: ignore
+
+
+def config_callback(ctx: typer.Context, param: typer.CallbackParam, value: str):
+    if value:
+        try:
+            with open(value, 'r') as f:
+                config = yaml.safe_load(f)
+
+            ctx.default_map = ctx.default_map or {}
+            ctx.default_map.update(config)
+        except Exception as ex:
+            raise typer.BadParameter(str(ex))
+
+    return value
+
+
+@app.command(rich_help_panel="Status commands")
+def systems(
+    name: Optional[str] = typer.Option(
+        None, "-n", "--name", help="Get information for only one system."
+    ),
+    raw: bool = typer.Option(False, "--json", "--raw", help="Print the output in json format."),
+    pager: Optional[bool] = typer.Option(
+        True, help="Display the output in a pager application."
```

Comment:
Not sure about this but, but perhaps this should be `False` since the option '--pager' does not take a value?